File: FlexViewPager/Sample/src/main/java/com/dreamgyf/android/ui/widget/viewpager/flex/MainActivity.java
package com.dreamgyf.android.ui.widget.viewpager.flex;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

	private FlexViewPager mVpFlex;

	private FlexPagerAdapter mAdapter;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);

		List<View> viewList = new ArrayList<>();
		viewList.add(LayoutInflater.from(this).inflate(R.layout.view_demo_1, null));
		viewList.add(LayoutInflater.from(this).inflate(R.layout.view_demo_2, null));
		viewList.add(LayoutInflater.from(this).inflate(R.layout.view_demo_3, null));
		viewList.add(LayoutInflater.from(this).inflate(R.layout.view_demo_4, null));
		viewList.add(LayoutInflater.from(this).inflate(R.layout.view_demo_5, null));

		mVpFlex = findViewById(R.id.vp_flex);
		mVpFlex.setAdapter(mAdapter = new FlexPagerAdapter());
		mAdapter.setViews(viewList);
	}
}

File: SpaceItemDecoration/Sample/src/main/java/com/dreamgyf/android/ui/rv/MainActivity.kt
package com.dreamgyf.android.ui.rv

import android.content.res.ColorStateList
import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private lateinit var rv: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rv = findViewById(R.id.rv)
        rv.adapter = object : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
            override fun onCreateViewHolder(
                parent: ViewGroup,
                viewType: Int
            ): RecyclerView.ViewHolder {
                val view = View(this@MainActivity)
                view.background = GradientDrawable().apply {
                    color = ColorStateList.valueOf(Color.GRAY)
                    cornerRadius = 50f
                }
                view.layoutParams = ViewGroup.LayoutParams(
                    ViewGroup.LayoutParams.MATCH_PARENT,
                    1000
                )
                return object : RecyclerView.ViewHolder(view) {}
            }

            override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
            }

            override fun getItemCount(): Int {
                return 10
            }

        }
        rv.layoutManager = GridLayoutManager(this, 3)
        rv.addItemDecoration(SpaceItemDecoration(50, 30))
    }
}